machinist/state/controlplane_test.go

```diff
@@ -0,0 +1,60 @@
+package state_test
+
+import (
+	"github.com/enfabrica/enkit/lib/srand"
+	"github.com/enfabrica/enkit/machinist/state"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"strconv"
+	"testing"
+)
+
+
+
+func TestReadInController(t *testing.T) {
+	rng := rand.New(srand.Source)
+	rngName := func() string {
+		return strconv.Itoa(rng.Int())
+	}
+	t.Run("Test consecutive read ins", func(t *testing.T) {
+		rname := rngName() + ".json"
```

Comment:
why not just use `TempFile`? and in the returned object, use the `Name()` method to discover the path? It'll help prevent spurious failures.

Reply:
If you do use TempFile, there's very little reason left to use the nested t.Run? so it could just be multiple top level `func Test...`?

Reply:
The issue is that os.TempFile actually creates the file. So if I use that, ReadInController always fails and I can't test auto state file creation.

Reply:
ack, makes sense!